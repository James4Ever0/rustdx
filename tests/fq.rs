use insta::assert_debug_snapshot;

#[test]
fn day_sz000001() -> rustdx::Result<()> {
    use rustdx::file::{
        day::Day,
        gbbq::{Fq, Gbbq},
    };
    let day_src = std::fs::read("tests/assets/sz000001.day")?;
    let days = day_src.chunks_exact(32).map(|arr| Day::from_bytes(1, arr));

    let mut gbbq_src = std::fs::read("tests/assets/gbbq")?;
    let stock_gbbq = Gbbq::filter_hashmap(Gbbq::iter(&mut gbbq_src[4..]));

    let fq = Fq::new(days, stock_gbbq.get(&1).unwrap()).unwrap();
    assert_debug_snapshot!(&fq[..3]);
    Ok(())
}
