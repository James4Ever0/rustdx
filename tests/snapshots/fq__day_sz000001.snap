---
source: tests/fq.rs
expression: "&fq[..3]"

---
[
    Fq {
        code: 1,
        date: 19910403,
        factor: 1.0,
        close: 49.0,
        preclose: 49.0,
        trading: true,
        xdxr: false,
    },
    Fq {
        code: 1,
        date: 19910404,
        factor: 0.9951020065619021,
        close: 48.7599983215332,
        preclose: 49.0,
        trading: true,
        xdxr: false,
    },
    Fq {
        code: 1,
        date: 19910405,
        factor: 0.9902040909747688,
        close: 48.52000045776367,
        preclose: 48.7599983215332,
        trading: true,
        xdxr: false,
    },
]
